src/use_case/send_notification/NotificationInteractor.java

```diff
@@ -0,0 +1,192 @@
+                    } else if (task.getDeadline().equals(plusTwo)) {
+                        if (!duePlusTwo.containsKey(project)) {
+                            duePlusTwo.put(project, new ArrayList<Task>());
+                        }
+                        duePlusTwo.get(project).add(task);
+                    // checks if the task is due today
+                    } else if (task.getDeadline().equals(date)) {
+                        if (!dueToday.containsKey(project)) {
+                            dueToday.put(project, new ArrayList<Task>());
+                        }
+                        dueToday.get(project).add(task);
+                    }
+                }
+            }
+        }
+        // checks that at least one of the HashMaps is non-empty
+        if (!duePlusOne.isEmpty() || !duePlusTwo.isEmpty() || !dueToday.isEmpty()) {
+            // generates gpt query and executes the api call using helper functions
+            String[] gptQuery = generateQuery(duePlusOne, duePlusTwo, dueToday);
+            String gptOutput = apiCall(gptQuery[0]);
+            // Creates output data and calls presenter
+            NotificationOutputData notificationOutputData = new NotificationOutputData(gptQuery[1] + gptOutput);
+            notificationPresenter.prepareNotificationView(notificationOutputData);
+        }
+
+
+
+    }
+    // Generates the string that will be used to query gpt
+    private static String[] generateQuery(HashMap<Project, ArrayList<Task>> one, HashMap<Project, ArrayList<Task>> two,
+                                        HashMap<Project, ArrayList<Task>> today) {
+        // initializes the StringBuilder and briefly explains to gpt the situation.
+        StringBuilder m = new StringBuilder("I am working on some projects right now and these are some " +
+                "upcoming tasks that are due. \n");
+        StringBuilder q = new StringBuilder();
+        // Note that all three if statements are essentially exactly the same as this one.
+        // They just add information about the tasks due on their specific days.
+        if (!today.isEmpty()) {
+            // Curly brackets, {}, are used to encompass tasks due on the same day and
+            // related to the same project.
+            q.append("These tasks are due today: {\n");
+            for (Project project : today.keySet()) {
+                // The project description is omitted to save space in the call. I aim to have
+                // gpt give a relatively brief response, and I feel that the task descriptions
+                // along with the project title should be enough information to give some
+                // quick advice and a motivational statement.
+                q.append("For Project: ").append(project.getName()).append(" {\n");
+                ArrayList<Task> tasks = today.get(project);
+                for (Task task : tasks) {
+                    q.append("Task Name: ").append(task.getName()).append(", ");
```

Comment:
I don't know if you chose it to add a comma instead of a next line character, but I guess it's a choice of preference when inputting it into gpt

Reply:
This applies for all three cases

Reply:
You should also check for an error from gpt